src/cargo/ops/cargo_new.rs

```diff
@@ -38,10 +44,71 @@ impl Decodable for VersionControl {
     }
 }
 
+#[derive(Clone, Debug, PartialEq)]
+pub enum License {
+    MIT,
+    BSD3,
+    APACHE2,
+    MPL2,
+    GPL3,
+}
+
+impl License {
+    fn write_file(&self, path: &Path, holders: &str) -> io::Result<()> {
+        let mut license_text: Vec<u8> = Vec::new();
+        let _ = self.license_text(&mut license_text, holders);
+        file(path, &license_text)
+    }
+
+    fn license_text<F: io::Write>(&self, f: &mut F, holders: &str) -> io::Result<()> {
```

Comment:
I think it'd be fine for this to just return a `String` instead of writing into a writer